store/src/java/com/zimbra/cs/filter/jsieve/HeaderTest.java

```diff
@@ -348,11 +348,13 @@ public static void evaluateVarExp(ZimbraMailAdapter mailAdapter, List<String> so
                 }
                 List<String> decodedValues = new ArrayList<>();
                 for (String value : values) {
+                    value = MimeUtility.unfold(value);
                     try {
-                        decodedValues.add(MimeUtility.decodeText(MimeUtility.unfold(value)));
+                        value = MimeUtility.decodeText(value);
                     } catch (UnsupportedEncodingException e) {
-                        throw new SieveMailException("Exception occured while decoding header value", e);
+                        // "value" would contain the undecoded value, fine
```

Comment:
Log a message regarding the exception

Reply:
As in the return comment on the Shrikant, the result status of the decoding is detectable by other log event.  Besides, most of other places where the UnsupportedEncodingException is cached don't log the exception.